src/Domain.Linnapps/Reports/ProductsOnHoldService.cs

```diff
@@ -32,20 +34,20 @@ public ResultsModel GetProductsOnHold()
             results.SetColumnType(4, GridDisplayType.TextValue);
             results.SetColumnType(5, GridDisplayType.TextValue);
 
-            foreach (var productOnHold in productsOnHold)
+            foreach (var salesArticleOnHold in salesArticlesOnHold)
             {
-                var row = results.AddRow(productOnHold.ArticleNumber.Replace("/", "%2F"), productOnHold.ArticleNumber);
-                results.SetGridTextValue(row.RowIndex, 0, productOnHold.SalesArticle.ArticleNumber);
-                results.SetGridTextValue(row.RowIndex, 1, productOnHold.SalesArticle.InvoiceDescription);
-                results.SetGridTextValue(row.RowIndex, 2, productOnHold.PutOnHoldByEmployee.FullName);
-                results.SetGridTextValue(row.RowIndex, 3, productOnHold.DateStarted.ToShortDateString());
-                results.SetGridTextValue(row.RowIndex, 4, productOnHold.AnticipatedEndDate != null ? ((DateTime) productOnHold.AnticipatedEndDate).ToShortDateString() : null);
-                results.SetGridTextValue(row.RowIndex, 5, productOnHold.ReasonStarted);
+                var holdStory = this.saHoldStoryRepository.FindById((int)salesArticleOnHold.LastHoldStoryId);
```

Comment:
what happens if the lastHoldStoryId is null?

Reply:
I don't think this can happen as the oracle logic gives sales articles a lastHoldStoryId whenever they are put on hold.